### ISSUE TITLE
StatsAndLogging thread reads the entire log into memory unnecessarily

### ISSUE THREAD

**Issue by joelarmstrong** [BD2KGenomics/toil#1594]
This can cause a leader crash when the log files are truly gigantic:

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/local/lib/python2.7/dist-packages/toil/statsAndLogging.py", line 129, in statsAndLoggingAggregator
    if jobStore.readStatsAndLogging(callback) == 0:
  File "/usr/local/lib/python2.7/dist-packages/toil/jobStores/aws/jobStore.py", line 627, in readStatsAndLogging
    for info in self._readStatsAndLogging(callback, self.statsFileOwnerID):
  File "/usr/local/lib/python2.7/dist-packages/toil/jobStores/aws/jobStore.py", line 650, in _readStatsAndLogging
    callback(readable)
  File "/usr/local/lib/python2.7/dist-packages/toil/statsAndLogging.py", line 121, in callback
    logWithFormatting(jobNames[0], messages)
  File "/usr/local/lib/python2.7/dist-packages/toil/statsAndLogging.py", line 117, in logWithFormatting
    'logging to hide this output\n%s', logFormat.join(jobLogs))
MemoryError
```

One could (correctly) argue my jobs shouldn't be generating 2 gigs of log files. But this is an easy fix, and the AWS job store takes pains to stream the log file, so the logger might as well stream each line as well. I have a simple patch that I'm testing, so I'm assigning this to myself, but I'm blocked on another issue right now.
